dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.SemanticKernel.Data;
+using NRedisStack.Search;
+
+namespace Microsoft.SemanticKernel.Connectors.Redis;
+
+/// <summary>
+/// Contains mapping helpers to use when creating a redis vector collection.
+/// </summary>
+internal static class RedisVectorStoreCollectionCreateMapping
+{
+    /// <summary>A set of number types that are supported for filtering.</summary>
+    public static readonly HashSet<Type> s_supportedFilterableNumericDataTypes =
+    [
+        typeof(short),
+        typeof(sbyte),
+        typeof(byte),
+        typeof(ushort),
+        typeof(int),
+        typeof(uint),
+        typeof(long),
+        typeof(ulong),
+        typeof(float),
+        typeof(double),
+        typeof(decimal),
+
+        typeof(short?),
+        typeof(sbyte?),
+        typeof(byte?),
+        typeof(ushort?),
+        typeof(int?),
+        typeof(uint?),
+        typeof(long?),
+        typeof(ulong?),
+        typeof(float?),
+        typeof(double?),
+        typeof(decimal?),
+    ];
+
+    /// <summary>
+    /// Map from the given list of <see cref="VectorStoreRecordProperty"/> items to the Redis <see cref="Schema"/>.
+    /// </summary>
+    /// <param name="properties">The property definitions to map from.</param>
+    /// <returns>The mapped Redis <see cref="Schema"/>.</returns>
+    /// <exception cref="InvalidOperationException">Thrown if there are missing required or unsupported configuration options set.</exception>
+    public static Schema MapToSchema(IEnumerable<VectorStoreRecordProperty> properties)
+    {
+        var schema = new Schema();
+
+        // Loop through all properties and create the index fields.
+        foreach (var property in properties)
+        {
+            // Key property.
+            if (property is VectorStoreRecordKeyProperty keyProperty)
+            {
+                // Do nothing, since key is not stored as part of the payload and therefore doesn't have to be added to the index.
```

Comment:
Should we go to the next iteration and start processing the next property using the `continue` statement/operator?

Reply:
Yep, makes sense.  Done.